packages/common/src/compiler/translator.ts

```diff
@@ -378,10 +380,14 @@ export const convertTable = (
                     startOfWeek,
                     true,
                 );
+
                 return {
                     ...acc,
-                    [subDimensionName]: additionalDim,
-                    ...processIntervalDimension(additionalDim),
+                    // When the additional dim is interval AND the base dimension is a interval base then we want to compute all additional dims with the parent intervals otherwise just set the additional dim
+                    ...(isInterval(additionalDim.type, column) &&
+                    dimension.isIntervalBase
+                        ? processIntervalDimension(additionalDim)
+                        : { [subDimensionName]: additionalDim }),
```

Comment:
This logic fixes 2 things:

1. Previously we were always trying to process interval dimensions on the additional dim + adding the additional dim as is, this was causing duplicates the duplicate mentioned in the ticket and the bug where we took into account the `time_intervals` property in the yaml for the additional dimension
2. Makes it so that if the base dimension is not an interval base (e.g. when base dimension is 'string' but somehow we want to add an additional dimension which is type 'date' - see mock data `MODEL_WITH_ADDITIONAL_DIMENSIONS`) then we won't generate intervals and the dimension will be just the raw value.

Overall this makes it so that:
1. if base dimension is date/time and my additional dimension is date/time then it will respect the intervals of the base dimension
2. if base dimension is date/time with intervals set to 'OFF' and my additional dimension is date/time, both will be raw value (no intervals)
3. if base dimension is NOT date/dime and additional dimension is date/time, my additional dimension will not have intervals
4. if base dimension is date/time and additional dimension is NOT date/time, my additional dimension will not have intervals

Point 3 can be changed to add the intervals to the additional dimension, but I believe that would be unexpected since they cannot be controlled via the yaml

Reply:
After the last commit this is no longer true, see description